apps/vscode/command_client/command_client.py

```diff
@@ -197,20 +198,30 @@ def run_command(
     return decoded_contents["returnValue"]
 
 
+command_server_directory = None
+
+
+# NB: See https://github.com/talonhub/community/issues/966 for why we do OS-specific temp dirs
+def get_platform_specific_communication_dir_path():
+    home_dir = Path(os.path.expanduser("~"))
+    directory_name = actions.user.command_server_directory()
+    if app.platform == "linux" or app.platform == "mac":
+        return home_dir / ".talon" / " .comms" / directory_name
```

Comment:
Don't snaps prevent access to `~/.*`?

Reply:
In the default containment profiles yes (I believe you need to request personal confinement to access dotfiles), but from what I read vscode should be okay due to it using classic confinement (which is effectively no confinement). See here from this same PR where I said similar: https://github.com/talonhub/community/pull/1362#discussion_r1460811407

@DonnerWolfBach said he was using VSCodium. I suspect since it's just an open source build of VSCode (?) and an IDE when packaged as a snap it would use the same classic confinement. Maybe if he's using the snap, he could confirm though.

From what I've seen they tend to grant classic confinement to all IDEs under the assumption they will be used to edit any files. For instance neovim snap also has [classic confinement](https://forum.snapcraft.io/t/classic-confinement-request-for-nvim/14405/17).

I think the only other option if we want to assume we need to communicate with a snaps with restricted confinement is /tmp, and the discussion on https://github.com/talonhub/community/issues/966 people couldn't agree, and it seemed some (pokey and rntz at least iirc) were leaning towards using $HOME over /tmp, so I just used $HOME for the sake of getting anything going.

Reply:
I don't use VSCodium in a snap unfortunately. I use the nix package